sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala

```diff
@@ -392,7 +392,7 @@ private[sql] trait SQLTestUtilsBase
    */
   protected def stripSparkFilter(df: DataFrame): DataFrame = {
     val schema = df.schema
-    val withoutFilters = df.queryExecution.sparkPlan.transform {
+    val withoutFilters = df.queryExecution.executedPlan.transform {
```

Comment:
Why making this change in this PR?

Reply:
Seems like the existing columnar logic within each plans have moved to `RowToColumnarExec` and `ColumnarToRowExec` to deduplicate but it's now dependent on `ApplyColumnarRulesAndInsertTransitions` rule, which, now, requires execution preparation (`QueryExecution.preparations`).

However, per the doc, `executedPlan` should be only used for execution ideally. It could have been best to avoid. @revans2 even though it's too late, can you please describe what does this PR fixes in the PR description (presumably by listing each item)?

I have no idea what this PR fixes from reading the PR description and JIRA.

Reply:
This PR replaces ColumnarBatchScan with ColumnarToRowExec.  This involved changes to all subclasses of COlumnarBatchScan and to AdaptiveSparkPlanExec so it would also execute the needed columnar transition rules.  I also made any fixes needed for tests.  I preferred to keep the changes as small as possible for the tests, which is why I made a small change here in a test utility class.

The issue was that some tests were directly execution the plan returned by this function, which used to work for some very limited use cases, but did not work in all cases.  I am happy to try and fix issues with this approach for the tests, just let me know what is the correct way to do it?

Reply:
I don't know the correct way to fix it for now - it needs some investigations. Can you clarify why it didn't work well?

The problem seems like by the internal behaviour changes - we now rely on `ApplyColumnarRulesAndInsertTransitions`. Can we make some investigation to confirm that it doesn't affect anything, and list up what changes were made in this PR description?

Reply:
Yes that is exactly what it is.

Reply:
@revans2, what I meant is like https://github.com/apache/spark/pull/25264 rather than  single line that describes what this PR proposes.